tools/devops/automation/templates/sign-and-notarized/dotnet-signing.yml

```diff
@@ -7,35 +7,93 @@ parameters:
 - name: isPR
   type: boolean
 
-steps:
-
-- template: setup.yml
-  parameters:
-    isPR: ${{ parameters.isPR }}
-
-- task: DownloadPipelineArtifact@2
-  displayName: Download not notaraized build
-  inputs:
-    artifact: 'not-signed-package'
-    patterns: '!*.pkg'
-    allowFailedBuilds: true
-    path: $(Build.SourcesDirectory)/package
-
-- ${{ if eq(parameters.isPR, false) }}:
-  - pwsh : |
-      # Get the list of files to sign
-      $msiFiles = Get-ChildItem -Path $(Build.SourcesDirectory)/package/ -Filter "*.msi"
-
-      # Add those files to an array
-      $SignFiles = @()
-      foreach($msi in $msiFiles) {
-          Write-Host "$($msi.FullName)"
-          $SignFiles += @{ "SrcPath"="$($msi.FullName)"}
+jobs:
+- job: configure
+  displayName: 'Configure build'
+  pool:
+    vmImage: windows-latest
+
+  steps:
+  - template: ../common/configure.yml
+
+- job: sign_notarize_dotnet
+  dependsOn:
+  - configure
+  displayName: 'Sign & Notarize Dotnet'
+  timeoutInMinutes: 1000
+  pool:
+    vmImage: internal-macos-11
+  workspace:
+    clean: all
+
+  steps:
+
+  - template: setup.yml
+    parameters:
+      isPR: ${{ parameters.isPR }}
+
+  - task: DownloadPipelineArtifact@2
+    displayName: Download not notaraized build
+    inputs:
+      artifact: 'not-signed-package'
+      patterns: '!*.pkg'
+      allowFailedBuilds: true
+      path: $(Build.SourcesDirectory)/package
+
+  - ${{ if eq(parameters.isPR, false) }}:
+    - pwsh : |
+        # Get the list of files to sign
+        $msiFiles = Get-ChildItem -Path $(Build.SourcesDirectory)/package/ -Filter "*.msi"
+
+        # Add those files to an array
+        $SignFiles = @()
+        foreach($msi in $msiFiles) {
+            Write-Host "$($msi.FullName)"
+            $SignFiles += @{ "SrcPath"="$($msi.FullName)"}
+        }
+
+        Write-Host "$msiFiles"
+
+        # array of dicts
+        $SignFileRecord = @(
+          @{
+            "Certs" = "400";
+            "SignFileList" = $SignFiles;
+          }
+        )
+
+        $SignFileList = @{
+            "SignFileRecordList" = $SignFileRecord
+        }
+
+        # Write the json to a file
+        ConvertTo-Json -InputObject $SignFileList -Depth 5 | Out-File -FilePath $(Build.ArtifactStagingDirectory)/MsiFiles2Notarize.json -Force
+        dotnet $Env:MBSIGN_APPFOLDER/ddsignfiles.dll /filelist:$(Build.ArtifactStagingDirectory)/MsiFiles2Notarize.json
+      displayName: 'Sign .msi'
+      condition: ${{ parameters.condition }}
```

Comment:
Do we distribute this MSI anywhere anymore? I know that in early .NET 6 previews we briefly built and shipped our own installers, but I don't think this is used anymore? This may be a good opportunity to remove this step and save some more time if it is no longer used.